modules/bioinspired/test/test_retina_ocl.cpp

```diff
@@ -82,48 +66,24 @@ PARAM_TEST_CASE(Retina_OCL, bool, int, bool, double, double)
         useLogSampling      = GET_PARAM(2);
         reductionFactor     = GET_PARAM(3);
         samplingStrength    = GET_PARAM(4);
-
-        if (!oclInit)
-        {
-            if (cv::ocl::haveOpenCL())
-            {
-                try
-                {
-                    const cv::ocl::DeviceInfo& dev = cv::ocl::Context::getContext()->getDeviceInfo();
-                    std::cout << "Device name:" << dev.deviceName << std::endl;
-                    oclAvailable = true;
-                }
-                catch (...)
-                {
-                    std::cout << "Device name: N/A" << std::endl;
-                }
-            }
-            oclInit = true;
-        }
     }
 };
 
-TEST_P(Retina_OCL, Accuracy)
+OCL_TEST_P(Retina_OCL, Accuracy)
 {
-    if (!oclAvailable)
-    {
-        std::cout << "SKIP test" << std::endl;
-        return;
-    }
-
-    using namespace cv;
     Mat input = imread(cvtest::TS::ptr()->get_data_path() + "shared/lena.png", colorMode);
     CV_Assert(!input.empty());
-    ocl::oclMat ocl_input(input);
 
-    Ptr<bioinspired::Retina> ocl_retina = bioinspired::createRetina_OCL(
+    OCL_ON();
+    Ptr<bioinspired::Retina> ocl_retina = bioinspired::createRetina(
         input.size(),
         colorMode,
         colorSamplingMethod,
         useLogSampling,
         reductionFactor,
         samplingStrength);
 
+    OCL_OFF();
     Ptr<bioinspired::Retina> gold_retina = bioinspired::createRetina(
         input.size(),
         colorMode,
```

Comment:
Only one creation scheme should be used and technically `ocl_retina` and `gold_retina` should be the same object or  similarly constructed (preferred). Please remove `OCL_ON`/`OCL_OFF`.